File: src/types/projects.ts
export type InitiativeStatus = 'planning' | 'on-track' | 'at-risk' | 'completed' | 'on-hold'

export type Initiative = {
  id: string
  name: string
  description: string
  owner: string
  status: InitiativeStatus
  phase: string
  progress: number
  budgetPlanned: number
  budgetSpent: number
  riskScore: number
  impactScore: number
  startDate: string
  dueDate: string
  dependencies: string[]
  tags: string[]
}

export const STATUS_META: Record<InitiativeStatus, { label: string; color: string }> = {
  planning: { label: 'Planning', color: 'var(--info)' },
  'on-track': { label: 'On Track', color: 'var(--success)' },
  'at-risk': { label: 'At Risk', color: 'var(--danger)' },
  completed: { label: 'Completed', color: 'var(--primary)' },
  'on-hold': { label: 'On Hold', color: 'var(--warning)' },
}

export const PHASES = ['Discovery', 'Foundation', 'Build', 'Migration', 'Optimization', 'Scale']

/** Composite health: high progress & impact good; high risk & budget overrun bad. */
export function healthScore(i: Initiative): number {
  const budgetVariance = i.budgetPlanned > 0 ? Math.max(0, (i.budgetSpent - i.budgetPlanned) / i.budgetPlanned) : 0
  const raw = i.progress * 0.4 + i.impactScore * 0.3 + (100 - i.riskScore) * 0.3 - budgetVariance * 40
  return Math.max(0, Math.min(100, Math.round(raw)))
}

export function budgetUtilization(i: Initiative): number {
  if (i.budgetPlanned <= 0) return 0
  return Math.round((i.budgetSpent / i.budgetPlanned) * 100)
}

export function formatCurrency(value: number): string {
  if (value >= 1_000_000) return `$${(value / 1_000_000).toFixed(2)}M`
  if (value >= 1_000) return `$${(value / 1_000).toFixed(0)}K`
  return `$${value}`
}

File: backend/app/schemas/projects.py
from pydantic import BaseModel, Field


class ProjectCreate(BaseModel):
    title: str
    description: str = ""
    status: str = "planning"
    risk_score: int = 0
    impact_score: int = 0
    health_score: int = 0
    budget_planned: float = 0
    budget_spent: float = 0
    progress: int = 0
    owner: str = ""
    phase: str = "Discovery"
    start_date: str = ""
    due_date: str = ""
    tags: list[str] = Field(default_factory=list)
    dependencies: list[str] = Field(default_factory=list)


class ProjectUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: str | None = None
    risk_score: int | None = None
    impact_score: int | None = None
    health_score: int | None = None
    budget_planned: float | None = None
    budget_spent: float | None = None
    progress: int | None = None
    owner: str | None = None
    phase: str | None = None
    start_date: str | None = None
    due_date: str | None = None
    tags: list[str] | None = None
    dependencies: list[str] | None = None


class ProjectResponse(BaseModel):
    id: str
    workspace_id: str
    name: str
    description: str
    owner: str
    status: str
    phase: str
    progress: int
    budgetPlanned: float
    budgetSpent: float
    riskScore: int
    impactScore: int
    startDate: str
    dueDate: str
    dependencies: list[str]
    tags: list[str]
